### PR TITLE
Overrides for tags and run_tags

### PR DESCRIPTION
**Pull request by guikcd**
### Summary

Support `tags` and/or `run_tags` in *overrides.auto.pkrvars.hcl*:
```
tags = {
  owner = "myteam"
  environment= "test"
}

run_tags = {
  owner = "packer"
  environment= "build"
}
```

Before this PR, users need to identity AMI and manually tags the resulted AMI.

Closes: #31

### Implementation details

Define a variable for tags and run_tags for overrides

Then for each variant:
* tags are moved into locals as default_tags
* merge the default + user provided tags and define tags
* define run_tags

### Testing

Building AMIs without overriding these tags are OK
Building AMIs with these tags are OK:

```
ALL=$(ls al*|sed "s/.pkr.hcl//")

for variant in $ALL
do
        rm -f manifest.json
        REGION=eu-west-1 make ${variant}
        ami=$(jq -r ".builds[0].artifact_id" manifest.json | awk -F':' '{print $NF}')
        # tags are set when AMI is available, not during creation
        echo "wait AMI"
        aws ec2 wait image-available --image-ids $ami
        snap=$(aws ec2 describe-images --region eu-west-1 --image-ids $ami --query "Images[0].BlockDeviceMappings[0].Ebs.SnapshotId" --output text)
        echo "check tags"
        aws ec2 describe-images --region eu-west-1 --image-ids $ami --query "Images[0].Tags[]" --output text | grep --quiet "myteam"
        if [ $? -ne 0 ]
        then
                echo "WARNING: tags are not set! for $variant / $ami"
        fi
        echo "unregister"
        aws ec2 deregister-image --image-id $ami
        echo "delete snap"
        aws ec2 delete-snapshot --snapshot-id $snap
done
```
New tests cover the changes: yes

### Description for the changelog

Feature: Support user provided tags and run_tags.

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: al2023neu.pkr.hcl [hunk 1]
```diff
@@ -1,5 +1,14 @@
 locals {
   ami_name_al2023neu = "${var.ami_name_prefix_al2023}-neuron-hvm-2023.0.${var.ami_version_al2023}${var.kernel_version_al2023}-x86_64"
+  tags = {
```
Comment (danehlim) [inline al2023neu.pkr.hcl:3]: Should this line read as `default_tags = {` (like most of the other AMI recipe files), or is this intentional?

File: al2kernel5dot10gpu.pkr.hcl [hunk 2]
```diff
@@ -1,5 +1,14 @@
 locals {
   ami_name_al2kernel5dot10gpu = "${var.ami_name_prefix_al2}-kernel-5.10-gpu-hvm-2.0.${var.ami_version_al2}-x86_64-ebs"
+  tags = {
```
Comment (danehlim) [inline al2kernel5dot10gpu.pkr.hcl:3]: Same comment as [here](https://github.com/aws/amazon-ecs-ami/pull/303/files#r1866566075).

File: al2.pkr.hcl [hunk 3]
```diff
@@ -29,14 +38,8 @@ source "amazon-ebs" "al2" {
...
   ssh_username  = "ec2-user"
-  tags = {
-    os_version          = "Amazon Linux 2"
-    source_image_name   = "{{ .SourceAMIName }}"
-    ecs_runtime_version = "Docker version ${var.docker_version}"
-    ecs_agent_version   = "${var.ecs_agent_version}"
-    ami_type            = "al2"
-    ami_version         = "2.0.${var.ami_version_al2}"
-  }
+  tags = "${local.merged_tags}"
```
Comment (danehlim) [inline al2.pkr.hcl:41]: You'll likely need to run `./packer fmt .` in `/amazon-ecs-ami/` directory to get pull request static checks to pass.

**Comment by danehlim** [comment]
Hi @guikcd, thank you for your opening this pull request! I have left a few minor comments/clarifying questions.

**Comment by guikcd** [comment]
I've fixed my errors.
I can't test the whole build of all image because of (ex.: variant al2023neu):

```
==> amazon-ebs.al2023neu: Provisioning with shell script: scripts/install-ecs-init.sh
==> amazon-ebs.al2023neu: + '[' -n '' ']'
==> amazon-ebs.al2023neu: + '[' -n '' ']'
==> amazon-ebs.al2023neu: ++ mktemp -d
==> amazon-ebs.al2023neu: + WORK_DIR=/tmp/tmp.JezVwldTwT
==> amazon-ebs.al2023neu: + trap 'rm -rf /tmp/tmp.JezVwldTwT' EXIT
==> amazon-ebs.al2023neu: + cat
==> amazon-ebs.al2023neu: + '[' -z '' ']'
==> amazon-ebs.al2023neu: ++ uname -m
==> amazon-ebs.al2023neu: + ARCH=x86_64
==> amazon-ebs.al2023neu: + host_suffix=
==> amazon-ebs.al2023neu: + grep -q '^cn-'
==> amazon-ebs.al2023neu: + ECS_INIT_URL=https://s3.eu-west-1.amazonaws.com/amazon-ecs-agent-eu-west-1/ecs-init-1.89.1-1.amzn2023.x86_64.rpm
==> amazon-ebs.al2023neu: + curl -fLSs -o /tmp/tmp.JezVwldTwT/ecs-init.rpm https://s3.eu-west-1.amazonaws.com/amazon-ecs-agent-eu-west-1/ecs-init-1.89.1-1.amzn2023.x86_64.rpm
==> amazon-ebs.al2023neu: curl: (22) The requested URL returned error: 403
==> amazon-ebs.al2023neu: + rm -rf /tmp/tmp.JezVwldTwT
==> amazon-ebs.al2023neu: Provisioning step had errors: Running the cleanup provisioner, if present...
```
The ecs-init is now archived and migrated to ecs-agent since September: https://github.com/aws/amazon-ecs-init

**Comment by danehlim** [comment]
Thanks @guikcd, the changes look good to me. I have approved this pull request.

The HTTP/1.1 403 Forbidden you are running into when trying to access https://s3.eu-west-1.amazonaws.com/amazon-ecs-agent-eu-west-1/ecs-init-1.89.1-1.amzn2023.x86_64.rpm is due to ecs-init-1.89.1-1 release still being ongoing at this time (it has not yet reached that bucket). Once that release is complete, it should be accessible.

**Review by danehlim — APPROVED** [review]

**Review by mye956 — APPROVED** [review]

**Pull request merged**